src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -479,34 +505,39 @@ private static boolean isDigit(int c) {
      * @return a decoded String
      */
     private String percentDecode(final String input) {
-        if (input == null) {
-            return null;
-        }
-        final String decoded = uriDecode(input);
-        if (!decoded.equals(input)) {
-            return decoded;
-        }
-        return input;
+        return uriDecode(input);
```

Comment:
If this is a private method that just calls another method with the same signature, why keep the private method around?

Reply:
Done. The latest comment removes the private uri methods.